Blurhash.ImageSharp/Decoder.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Linq;
+using System.Net.Mime;
+using Blurhash.Core;
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.PixelFormats;
+
+namespace Blurhash.ImageSharp
+{
+    public class Decoder : CoreDecoder
+    {
+        /// <summary>
+        /// Decodes a Blurhash string into a <c>System.Drawing.Image</c>
+        /// </summary>
+        /// <param name="blurhash">The blurhash string to decode</param>
+        /// <param name="outputWidth">The desired width of the output in pixels</param>
+        /// <param name="outputHeight">The desired height of the output in pixels</param>
+        /// <param name="punch">A value that affects the contrast of the decoded image. 1 means normal, smaller values will make the effect more subtle, and larger values will make it stronger.</param>
+        /// <returns>The decoded preview</returns>
+        public Image<Rgb24> Decode(string blurhash, int outputWidth, int outputHeight, double punch = 1.0)
+        {
+            var pixelData = base.CoreDecode(blurhash, outputWidth, outputHeight, punch);
+            return ConvertToBitmap(pixelData);
+        }
+
+        /// <summary>
+        /// Converts the library-independent representation of pixels into a bitmap
+        /// </summary>
+        /// <param name="pixelData">The library-independent representation of the image</param>
+        /// <returns>A <c>System.Drawing.Bitmap</c> in 32bpp-RGB representation</returns>
+        internal static Image<Rgb24> ConvertToBitmap(Blurhash.Core.Pixel[,] pixelData)
+        {
+            var width = pixelData.GetLength(0);
+            var height = pixelData.GetLength(1);
+
+            var data = Enumerable.Range(0, height)
+                .SelectMany(y => Enumerable.Range(0, width).Select(x => (x, y)))
```

Comment:
Why not a for loop? This linq expression is a perf killer.

Reply:
It was copied from another implementation, I'll expand it, as I think perf is key if it's to be used in a Xamarin app for example, which is my primary use-case.